### PR TITLE
feat: Allow setting the engine and compiler and the configuration

### LINKED ISSUE #104: Option to select backend
### Motivation

See also [Wasmer issue 605](https://github.com/wasmerio/wasmer/issues/605) : It would be great if it would be possible to dynamically choose different back-ends, instead of defaulting to Cranelift. For instance, the LLVM back-end might be preferred for the speed of execution of the output, and also features like SIMD that are currently exclusive to LLVM (though for the latter, SIMD would first need to be exposed through the C API, see [here](https://github.com/wasmerio/wasmer/issues/1108)).

### Proposed solution

Expose compile_with - first in C API, then in go-ext-wasm.

### PR DESCRIPTION
**Pull request by jubianchi**
Closes #104

### CODE REVIEW

**Comment by jubianchi** [comment]
bors try

File: wasmer/config.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+// UseSinglepassCompiler sets the compiler to Singlepass in the configuration.
+//
+//   config := NewConfig()
+//	 config.UseSinglepassCompiler()
+//
+func (self *Config) UseLLVMCompiler() {
+	C.wasm_config_set_engine(self._inner, C.LLVM)
+}
```
Comment (Hywan) [inline wasmer/config.go:60]: The documentation mentions the Singlepass compiler, but it sets the LLVM compiler. One method is missing, or the doc' is wrong :-p.

**Review by Hywan — COMMENTED** [review]
Apart from my comment, are we sure the wasmer-c-api shared lib comes with all the engines and all the compilers? I reckon it would be best to implement a `wasmer_available_compilers`-ish function in the C API to be sure a given compiler is available. Thoughts?

**Comment by jubianchi** [comment]
@Hywan for now on the engines/compilers, I only added the method for the ones I got working with the standard Wasmer buid.

I think it could be usefull to add something directly in Wasmer, yes, like a feature managed `panic` on not-available engines and compilers.

**Pull request merged**